### ISSUE TITLE
autotools configure script does not discern AVX512 hardware support

### ISSUE THREAD

**Issue by jrwrigh** [zlib-ng/zlib-ng#1559]
My CPU supports AVX2, but not AVX512. If I run configure with gcc 10.2.0, the configure command thinks that AVX512 should work (as the gcc does not complain on the test file), but in actuality it doesn't.

Here's the output of configure and make:
```shell
$ ./configure --zlib-compat
Checking for compiler... gcc
Checking for Symbol versioning... Yes.
Checking for shared library support... Building shared library libz.so.1.2.13.zlib-ng with gcc.
Checking for off64_t... Yes.
Checking for fseeko... Yes.
Checking for posix_memalign... Yes.
Checking for aligned_alloc... Yes.
Checking for strerror... Yes.
Checking for getauxval() or elf_aux_info() in sys/auxv.h... Yes.
Checking for unistd.h... Yes.
Checking for ptrdiff_t... Yes.
Checking for ANSI C compliant compiler...  Yes.
Checking for -fno-semantic-interposition... Yes.
Checking for -fno-lto... Yes.
Checking for attribute(visibility(hidden)) support... Yes.
Checking for attribute(visibility(internal)) support... Yes.
Checking for attribute(aligned) ... Yes.
Checking for _Thread_local ... Yes.
Checking for __builtin_ctz ... Yes.
Checking for __builtin_ctzll ... Yes.
Checking for AVX2 intrinsics ... Yes.
Checking for AVX512 intrinsics ... Yes.
Checking for AVX512 k-mask intrinsics ... Yes.
Check whether -mtune=cascadelake works ... Yes.
Checking for AVX512VNNI intrinsics ... Yes.
Checking for SSE4.2 intrinsics ... Yes.
Checking for SSE2 intrinsics ... Yes.
Checking for SSSE3 intrinsics ... Yes.
Checking for PCLMULQDQ intrinsics ... Yes.
Checking for VPCLMULQDQ intrinsics ... Yes.
Checking for XSAVE intrinsics ... No.
Uname: Linux
ARCH: x86_64
Using arch directory: arch/x86
Architecture-specific static object files: x86_features.o slide_hash_avx2.o chunkset_avx2.o compare256_avx2.o adler32_avx2.o adler32_avx512.o adler32_avx512_vnni.o adler32_sse42.o insert_string_sse42.o chunkset_sse2.o compare256_sse2.o slide_hash_sse2.o adler32_ssse3.o chunkset_sss
e3.o crc32_pclmulqdq.o crc32_vpclmulqdq.o
Architecture-specific shared object files: x86_features.lo slide_hash_avx2.lo chunkset_avx2.lo compare256_avx2.lo adler32_avx2.lo adler32_avx512.lo adler32_avx512_vnni.lo adler32_sse42.lo insert_string_sse42.lo chunkset_sse2.lo compare256_sse2.lo slide_hash_sse2.lo adler32_ssse3.lo
 chunkset_ssse3.lo crc32_pclmulqdq.lo crc32_vpclmulqdq.lo
 
 $ make -j
gcc -O2  -std=c11 -Wall -DNDEBUG -DHAVE_SYMVER -D_LARGEFILE64_SOURCE=1 -DHAVE_POSIX_MEMALIGN -DHAVE_ALIGNED_ALLOC -DHAVE_SYS_AUXV_H -DZLIB_COMPAT -DWITH_GZFILEOP -DHAVE_VISIBILITY_HIDDEN -DHAVE_VISIBILITY_INTERNAL -DHAVE_ATTRIBUTE_ALIGNED -DHAVE_THREAD_LOCAL -DHAVE_BUILTIN_CTZ -DHA
VE_BUILTIN_CTZLL -DX86_FEATURES -DX86_AVX2 -DX86_AVX512 -DX86_MASK_INTRIN -DX86_AVX512VNNI -DX86_SSE42 -DX86_SSE2 -DX86_SSSE3 -DX86_PCLMULQDQ_CRC -DX86_VPCLMULQDQ_CRC -DWITH_GZFILEOP -I/users/jrwrigh/software/zlib-ng -c -o example.o /users/jrwrigh/software/zlib-ng/test/example.c
gcc -O2  -std=c11 -Wall -DNDEBUG -DHAVE_SYMVER -D_LARGEFILE64_SOURCE=1 -DHAVE_POSIX_MEMALIGN -DHAVE_ALIGNED_ALLOC -DHAVE_SYS_AUXV_H -DZLIB_COMPAT -DWITH_GZFILEOP -DHAVE_VISIBILITY_HIDDEN -DHAVE_VISIBILITY_INTERNAL -DHAVE_ATTRIBUTE_ALIGNED -DHAVE_THREAD_LOCAL -DHAVE_BUILTIN_CTZ -DHA
VE_BUILTIN_CTZLL -DX86_FEATURES -DX86_AVX2 -DX86_AVX512 -DX86_MASK_INTRIN -DX86_AVX512VNNI -DX86_SSE42 -DX86_SSE2 -DX86_SSSE3 -DX86_PCLMULQDQ_CRC -DX86_VPCLMULQDQ_CRC -DWITH_GZFILEOP -I/users/jrwrigh/software/zlib-ng -c -o gzread.o /users/jrwrigh/software/zlib-ng/gzread.c
make -C arch/x86 adler32_avx512.o
make -C arch/x86 adler32_avx512_vnni.o
make -C arch/x86 crc32_vpclmulqdq.o
gcc -O2  -std=c11 -Wall -DNDEBUG -DHAVE_SYMVER -D_LARGEFILE64_SOURCE=1 -DHAVE_POSIX_MEMALIGN -DHAVE_ALIGNED_ALLOC -DHAVE_SYS_AUXV_H -DZLIB_COMPAT -DWITH_GZFILEOP -DHAVE_VISIBILITY_HIDDEN -DHAVE_VISIBILITY_INTERNAL -DHAVE_ATTRIBUTE_ALIGNED -DHAVE_THREAD_LOCAL -DHAVE_BUILTIN_CTZ -DHA
VE_BUILTIN_CTZLL -DX86_FEATURES -DX86_AVX2 -DX86_AVX512 -DX86_MASK_INTRIN -DX86_AVX512VNNI -DX86_SSE42 -DX86_SSE2 -DX86_SSSE3 -DX86_PCLMULQDQ_CRC -DX86_VPCLMULQDQ_CRC -DWITH_GZFILEOP -I/users/jrwrigh/software/zlib-ng -c -o minigzip.o /users/jrwrigh/software/zlib-ng/test/minigzip.c
make[1]: Entering directory '/users/jrwrigh/software/zlib-ng/arch/x86'
gcc -O2  -std=c11 -Wall -DNDEBUG -DHAVE_SYMVER -D_LARGEFILE64_SOURCE=1 -DHAVE_POSIX_MEMALIGN -DHAVE_ALIGNED_ALLOC -DHAVE_SYS_AUXV_H -DZLIB_COMPAT -DWITH_GZFILEOP -DHAVE_VISIBILITY_HIDDEN -DHAVE_VISIBILITY_INTERNAL -DHAVE_ATTRIBUTE_ALIGNED -DHAVE_THREAD_LOCAL -DHAVE_BUILTIN_CTZ -DHA
VE_BUILTIN_CTZLL -DX86_FEATURES -DX86_AVX2 -DX86_AVX512 -DX86_MASK_INTRIN -DX86_AVX512VNNI -DX86_SSE42 -DX86_SSE2 -DX86_SSSE3 -DX86_PCLMULQDQ_CRC -DX86_VPCLMULQDQ_CRC -mavx512f -mavx512dq -mavx512bw -mavx512vl -mtune=cascadelake -fno-lto -I/users/jrwrigh/software/zlib-ng/arch/x86 -
I/users/jrwrigh/software/zlib-ng -c -o adler32_avx512.o /users/jrwrigh/software/zlib-ng/arch/x86/adler32_avx512.c
make[1]: Entering directory '/users/jrwrigh/software/zlib-ng/arch/x86'
gcc -O2  -std=c11 -Wall -DNDEBUG -DHAVE_SYMVER -D_LARGEFILE64_SOURCE=1 -DHAVE_POSIX_MEMALIGN -DHAVE_ALIGNED_ALLOC -DHAVE_SYS_AUXV_H -DZLIB_COMPAT -DWITH_GZFILEOP -DHAVE_VISIBILITY_HIDDEN -DHAVE_VISIBILITY_INTERNAL -DHAVE_ATTRIBUTE_ALIGNED -DHAVE_THREAD_LOCAL -DHAVE_BUILTIN_CTZ -DHA
VE_BUILTIN_CTZLL -DX86_FEATURES -DX86_AVX2 -DX86_AVX512 -DX86_MASK_INTRIN -DX86_AVX512VNNI -DX86_SSE42 -DX86_SSE2 -DX86_SSSE3 -DX86_PCLMULQDQ_CRC -DX86_VPCLMULQDQ_CRC -mpclmul -msse4.2 -mvpclmulqdq -mavx512f -mavx512f -mavx512dq -mavx512bw -mavx512vl -mtune=cascadelake -fno-lto -I/
users/jrwrigh/software/zlib-ng/arch/x86 -I/users/jrwrigh/software/zlib-ng -c -o crc32_vpclmulqdq.o /users/jrwrigh/software/zlib-ng/arch/x86/crc32_vpclmulqdq.c
gcc -O2  -std=c11 -Wall -fPIC -DNDEBUG -DHAVE_SYMVER -D_LARGEFILE64_SOURCE=1 -DHAVE_POSIX_MEMALIGN -DHAVE_ALIGNED_ALLOC -DHAVE_SYS_AUXV_H -DZLIB_COMPAT -DWITH_GZFILEOP -fno-semantic-interposition -DHAVE_VISIBILITY_HIDDEN -DHAVE_VISIBILITY_INTERNAL -DHAVE_ATTRIBUTE_ALIGNED -DHAVE_TH
READ_LOCAL -DHAVE_BUILTIN_CTZ -DHAVE_BUILTIN_CTZLL -DX86_FEATURES -DX86_AVX2 -DX86_AVX512 -DX86_MASK_INTRIN -DX86_AVX512VNNI -DX86_SSE42 -DX86_SSE2 -DX86_SSSE3 -DX86_PCLMULQDQ_CRC -DX86_VPCLMULQDQ_CRC -DPIC -DWITH_GZFILEOP -I/users/jrwrigh/software/zlib-ng -c -o gzread.lo /users/jr
wrigh/software/zlib-ng/gzread.c
make[1]: Entering directory '/users/jrwrigh/software/zlib-ng/arch/x86'
gcc -O2  -std=c11 -Wall -DNDEBUG -DHAVE_SYMVER -D_LARGEFILE64_SOURCE=1 -DHAVE_POSIX_MEMALIGN -DHAVE_ALIGNED_ALLOC -DHAVE_SYS_AUXV_H -DZLIB_COMPAT -DWITH_GZFILEOP -DHAVE_VISIBILITY_HIDDEN -DHAVE_VISIBILITY_INTERNAL -DHAVE_ATTRIBUTE_ALIGNED -DHAVE_THREAD_LOCAL -DHAVE_BUILTIN_CTZ -DHA
VE_BUILTIN_CTZLL -DX86_FEATURES -DX86_AVX2 -DX86_AVX512 -DX86_MASK_INTRIN -DX86_AVX512VNNI -DX86_SSE42 -DX86_SSE2 -DX86_SSSE3 -DX86_PCLMULQDQ_CRC -DX86_VPCLMULQDQ_CRC -mavx512f -mavx512dq -mavx512bw -mavx512vl -mavx512vnni -mtune=cascadelake -fno-lto -I/users/jrwrigh/software/zlib-
ng/arch/x86 -I/users/jrwrigh/software/zlib-ng -c -o adler32_avx512_vnni.o /users/jrwrigh/software/zlib-ng/arch/x86/adler32_avx512_vnni.c
make -C arch/x86 adler32_avx512.lo
make -C arch/x86 adler32_avx512_vnni.lo
make -C arch/x86 crc32_vpclmulqdq.lo
make[1]: Entering directory '/users/jrwrigh/software/zlib-ng/arch/x86'
gcc -O2  -std=c11 -Wall -fPIC -DNDEBUG -DHAVE_SYMVER -D_LARGEFILE64_SOURCE=1 -DHAVE_POSIX_MEMALIGN -DHAVE_ALIGNED_ALLOC -DHAVE_SYS_AUXV_H -DZLIB_COMPAT -DWITH_GZFILEOP -fno-semantic-interposition -DHAVE_VISIBILITY_HIDDEN -DHAVE_VISIBILITY_INTERNAL -DHAVE_ATTRIBUTE_ALIGNED -DHAVE_TH
READ_LOCAL -DHAVE_BUILTIN_CTZ -DHAVE_BUILTIN_CTZLL -DX86_FEATURES -DX86_AVX2 -DX86_AVX512 -DX86_MASK_INTRIN -DX86_AVX512VNNI -DX86_SSE42 -DX86_SSE2 -DX86_SSSE3 -DX86_PCLMULQDQ_CRC -DX86_VPCLMULQDQ_CRC -mavx512f -mavx512dq -mavx512bw -mavx512vl -mtune=cascadelake -fno-lto -DPIC -I/u
sers/jrwrigh/software/zlib-ng/arch/x86 -I/users/jrwrigh/software/zlib-ng -c -o adler32_avx512.lo /users/jrwrigh/software/zlib-ng/arch/x86/adler32_avx512.c
make[1]: Entering directory '/users/jrwrigh/software/zlib-ng/arch/x86'
gcc -O2  -std=c11 -Wall -fPIC -DNDEBUG -DHAVE_SYMVER -D_LARGEFILE64_SOURCE=1 -DHAVE_POSIX_MEMALIGN -DHAVE_ALIGNED_ALLOC -DHAVE_SYS_AUXV_H -DZLIB_COMPAT -DWITH_GZFILEOP -fno-semantic-interposition -DHAVE_VISIBILITY_HIDDEN -DHAVE_VISIBILITY_INTERNAL -DHAVE_ATTRIBUTE_ALIGNED -DHAVE_TH
READ_LOCAL -DHAVE_BUILTIN_CTZ -DHAVE_BUILTIN_CTZLL -DX86_FEATURES -DX86_AVX2 -DX86_AVX512 -DX86_MASK_INTRIN -DX86_AVX512VNNI -DX86_SSE42 -DX86_SSE2 -DX86_SSSE3 -DX86_PCLMULQDQ_CRC -DX86_VPCLMULQDQ_CRC -mpclmul -msse4.2 -mvpclmulqdq -mavx512f -mavx512f -mavx512dq -mavx512bw -mavx512
vl -mtune=cascadelake -fno-lto -DPIC -I/users/jrwrigh/software/zlib-ng/arch/x86 -I/users/jrwrigh/software/zlib-ng -c -o crc32_vpclmulqdq.lo /users/jrwrigh/software/zlib-ng/arch/x86/crc32_vpclmulqdq.c
make[1]: Entering directory '/users/jrwrigh/software/zlib-ng/arch/x86'
gcc -O2  -std=c11 -Wall -fPIC -DNDEBUG -DHAVE_SYMVER -D_LARGEFILE64_SOURCE=1 -DHAVE_POSIX_MEMALIGN -DHAVE_ALIGNED_ALLOC -DHAVE_SYS_AUXV_H -DZLIB_COMPAT -DWITH_GZFILEOP -fno-semantic-interposition -DHAVE_VISIBILITY_HIDDEN -DHAVE_VISIBILITY_INTERNAL -DHAVE_ATTRIBUTE_ALIGNED -DHAVE_TH
READ_LOCAL -DHAVE_BUILTIN_CTZ -DHAVE_BUILTIN_CTZLL -DX86_FEATURES -DX86_AVX2 -DX86_AVX512 -DX86_MASK_INTRIN -DX86_AVX512VNNI -DX86_SSE42 -DX86_SSE2 -DX86_SSSE3 -DX86_PCLMULQDQ_CRC -DX86_VPCLMULQDQ_CRC -mavx512f -mavx512dq -mavx512bw -mavx512vl -mavx512vnni -mtune=cascadelake -fno-l
to -DPIC -I/users/jrwrigh/software/zlib-ng/arch/x86 -I/users/jrwrigh/software/zlib-ng -c -o adler32_avx512_vnni.lo /users/jrwrigh/software/zlib-ng/arch/x86/adler32_avx512_vnni.c
make[1]: Leaving directory '/users/jrwrigh/software/zlib-ng/arch/x86'
cp arch/x86/adler32_avx512.lo adler32_avx512.lo
/tmp/ccgsYWa9.s: Assembler messages:
/tmp/ccgsYWa9.s:68: Error: no such instruction: `vpdpbusd %zmm7,%zmm4,%zmm2'
/tmp/ccgsYWa9.s:70: Error: no such instruction: `vpdpbusd %zmm7,%zmm5,%zmm3'
/tmp/ccgsYWa9.s:124: Error: no such instruction: `vpdpbusd %zmm7,%zmm2,%zmm3'
/tmp/ccJa0i49.s: Assembler messages:
/tmp/ccJa0i49.s:68: Error: no such instruction: `vpdpbusd %zmm7,%zmm4,%zmm2'
/tmp/ccJa0i49.s:70: Error: no such instruction: `vpdpbusd %zmm7,%zmm5,%zmm3'
/tmp/ccgsYWa9.s:218: Error: no such instruction: `vpdpbusd %ymm7,%ymm3,%ymm2'
/tmp/ccgsYWa9.s:222: Error: no such instruction: `vpdpbusd %ymm7,%ymm4,%ymm5'
/tmp/ccJa0i49.s:124: Error: no such instruction: `vpdpbusd %zmm7,%zmm2,%zmm3'
/tmp/ccgsYWa9.s:275: Error: no such instruction: `vpdpbusd %ymm7,%ymm2,%ymm5'
/tmp/ccJa0i49.s:218: Error: no such instruction: `vpdpbusd %ymm7,%ymm3,%ymm2'
/tmp/ccJa0i49.s:222: Error: no such instruction: `vpdpbusd %ymm7,%ymm4,%ymm5'
/tmp/ccJa0i49.s:275: Error: no such instruction: `vpdpbusd %ymm7,%ymm2,%ymm5'
Makefile:126: recipe for target 'adler32_avx512_vnni.lo' failed
Makefile:123: recipe for target 'adler32_avx512_vnni.o' failed
make[1]: *** [adler32_avx512_vnni.lo] Error 1
make[1]: *** [adler32_avx512_vnni.o] Error 1
make[1]: Leaving directory '/users/jrwrigh/software/zlib-ng/arch/x86'
make[1]: Leaving directory '/users/jrwrigh/software/zlib-ng/arch/x86'
Makefile:167: recipe for target 'arch/x86/adler32_avx512_vnni.o' failed
make: *** [arch/x86/adler32_avx512_vnni.o] Error 2
make: *** Waiting for unfinished jobs....
Makefile:170: recipe for target 'arch/x86/adler32_avx512_vnni.lo' failed
make: *** [arch/x86/adler32_avx512_vnni.lo] Error 2
make[1]: Leaving directory '/users/jrwrigh/software/zlib-ng/arch/x86'
/tmp/ccIW4j88.s: Assembler messages:
/tmp/ccIW4j88.s:325: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:326: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:343: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:344: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:345: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:346: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:347: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:348: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:349: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:350: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:366: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:367: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:370: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:371: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:373: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:374: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:687: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:688: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:711: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:712: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:713: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:714: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:715: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:716: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:717: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:718: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:746: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:747: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:749: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:750: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:752: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccIW4j88.s:753: Error: operand type mismatch for `vpclmulqdq'
Makefile:96: recipe for target 'crc32_vpclmulqdq.lo' failed
make[1]: *** [crc32_vpclmulqdq.lo] Error 1
make[1]: Leaving directory '/users/jrwrigh/software/zlib-ng/arch/x86'
Makefile:170: recipe for target 'arch/x86/crc32_vpclmulqdq.lo' failed
make: *** [arch/x86/crc32_vpclmulqdq.lo] Error 2
/tmp/ccG1iQ19.s: Assembler messages:
/tmp/ccG1iQ19.s:325: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:326: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:343: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:344: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:345: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:346: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:347: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:348: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:349: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:350: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:366: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:367: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:370: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:371: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:373: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:374: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:686: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:687: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:710: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:711: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:712: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:713: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:714: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:715: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:716: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:717: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:745: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:746: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:748: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:749: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:751: Error: operand type mismatch for `vpclmulqdq'
/tmp/ccG1iQ19.s:752: Error: operand type mismatch for `vpclmulqdq'
Makefile:93: recipe for target 'crc32_vpclmulqdq.o' failed
make[1]: *** [crc32_vpclmulqdq.o] Error 1
make[1]: Leaving directory '/users/jrwrigh/software/zlib-ng/arch/x86'
Makefile:167: recipe for target 'arch/x86/crc32_vpclmulqdq.o' failed
make: *** [arch/x86/crc32_vpclmulqdq.o] Error 2
```

I can attach the `configure.log` if you like.

**Comment by phprus** [comment]
Your ``binutils`` is too old for gcc-10.

**Comment by jrwrigh** [comment]
I'm not sure `binutils` (or really `as`, which I assume is the culprit you're referring to) is to blame here. Correct, there is a mismatch between `as` and `gcc` in this case. But all the `configure` script is doing compile a short program (using GCC), does so successfully, and (wrongly) thinks it can use AVX512 instructions. So it seems like the `configure` script failed to catch that building AVX512 was never going to work.

**Comment by haampie** [comment]
The issue is that the assembler never gets to see the AVX512 instructions because they're optimized out with `-O2`.

So, the only thing tested for is whether the compiler supports AVX512.

A better test would not hurt.

**Comment by haampie** [comment]
Fix in #1562
